src/Microsoft.AspNetCore.Sockets/HttpConnectionDispatcher.cs

```diff
@@ -316,31 +321,51 @@ private Task ProcessNegotiate(HttpContext context)
                 return;
             }
 
-            // Collect the message and write it to the channel
-            // TODO: Need to use some kind of pooled memory here.
+            // Read the entire payload to a byte array for now because Pipelines and ReadOnlyBytes
+            // don't play well with each other yet.
             byte[] buffer;
             using (var stream = new MemoryStream())
             {
                 await context.Request.Body.CopyToAsync(stream);
+                await stream.FlushAsync();
```

Comment:
Yes indeed it is! I puzzled over why I was getting an empty array out of `ToArray` for a few minutes before I realized it was required :).